### PR TITLE
fix(scale): accept protocol-correct Acaia weight frames

### PR DESCRIPTION
**Pull request by ODevStudio**
## Summary

- Problem: protocol-correct Acaia Lunar/Pyxis frames were rejected because declared lengths and record tails were validated too narrowly.
- Why it matters: initialization could miss valid weight/settings notifications and repeatedly disconnect.
- What changed: accept real settings frames, short heartbeat buttons, bit-field weight flags, and complete weights with unknown/incomplete trailing records while preserving frame isolation.
- What did NOT change: readiness still requires a decoded valid weight; strict checksum rejection was not added; physical hardware validation remains outstanding.

## Change Type

- [x] Bug fix
- [ ] Feature
- [ ] Refactor required for the fix
- [ ] Docs
- [ ] Security hardening
- [ ] Chore / infra
- [ ] Plugin (DYE2 or bundled skin)

## Scope

- [x] BLE transport / device comms
- [ ] REST API / handlers
- [ ] WebSocket API
- [ ] Machine state / shot logic
- [x] Scale / weight / flow
- [ ] Profiles / beans / grinders / workflows
- [ ] WebUI skins
- [ ] Plugins / JS runtime
- [ ] UI / Flutter widgets
- [ ] Storage / Drift database
- [ ] CI / build / infra
- [ ] Docs / specs

## Independent Protocol Alignment

- The timer notification registration argument changed from `0x01` to `0x05`. It controls the requested number of heartbeats between timer notifications.
- Both maintained/reference implementations use `0x05`: [`encode_notification_request()` in aioacaia](https://github.com/zweckj/aioacaia/blob/main/aioacaia/encoder.py) and [`encodeNotificationRequest()` in pyacaia](https://github.com/lucapinello/pyacaia/blob/master/pyacaia/__init__.py).
- Focused unit coverage asserts the registration payload. Physical Lunar/Pyxis beta validation remains outstanding.
- This alignment is independent of and not required for the declared-length parser correction.

## Linked Issues

- Related #509
- Regression from #528

## Root Cause

- Root cause: Acaia byte 3 declares the span beginning at the length byte, including event and semantic records but excluding checksum bytes. The parser compared it to inner body sizes and required every trailing record to be recognized.
- Missing detection or guardrail: settings required exactly 3; all heartbeat events required 8; weight flags were treated as a scalar; partial-frame recovery could consume the next frame header as a flag byte.
- Contributing context (if known): real aioacaia captures use declared-length-9 settings frames, declared-length-6 button heartbeats, combined flag bits, and optional/unknown trailing records.

## Regression Test Plan

- Coverage level that should have caught this:
  - [x] Unit test
  - [x] Integration test (mock transport edge)
  - [ ] End-to-end test (simulate=1 + curl/websocat)
  - [ ] Existing coverage already sufficient
- Target test or file: `test/unit/models/acaia_scale_test.dart`
- Scenario the test should lock in: real settings battery 93, short heartbeat button liveness, flags 0x03 negative weight, direct and heartbeat weights with unknown/incomplete tails, and recovery after truncated frames.
- If no new test added, why not: New tests were added.

## Documentation Obligations

- [ ] API spec updated: `assets/api/rest_v1.yml` or `assets/api/websocket_v1.yml` (if REST/WebSocket changed)
- [ ] API docs updated: `doc/Api.md` (if user-facing endpoint changed)
- [ ] Plugin docs updated: `doc/Plugins.md` (if events/API changed)
- [ ] Skin docs updated: `doc/Skins.md` (if skin behavior changed)
- [ ] Profile docs updated: `doc/Profiles.md` (if profile handling changed)
- [ ] Device docs updated: `doc/DeviceManagement.md` (if device flows changed)
- [x] N/A - no docs affected

## Security Impact

- New or changed REST endpoints? No
- New or changed WebSocket topics? No
- New or changed network calls? No
- BLE/USB surface changed? Yes - notification parsing is more compatible, but remains bounded by declared length and known body validation.
- File system access changed? No
- Plugin sandbox boundary changed? No
- If any Yes, explain risk and mitigation: The parser processes only declared frame payloads; boundary-adjacent headers are used only for bounded resynchronization, incomplete weight bodies are rejected, and strict structural validation remains for non-weight heartbeat records.

## User-Visible Changes

Acaia Lunar/Pyxis scales can remain initialized when they send protocol-correct settings, button, or weight frame variants that were previously rejected.

### Local gates

- [x] Direct Dart formatter on both changed files - no remaining formatting changes
- [x] `flutter analyze --no-pub` - no issues found
- [x] `flutter test test/unit/models/acaia_scale_test.dart` - all 34 tests passed
- [ ] `flutter test --no-pub` - 2,581 passed; 14 unrelated environment/asset failures remain: QuickJS DLL loading, onboarding cascade, and missing bundled-skins manifest
- [ ] `(cd packages/dye2-plugin && npm run build)` - not run; no plugin files changed

### Manual verification

- OS / platform tested: Windows
- Simulated devices? (simulate=1): No
- Real hardware? (DE1/Bengle/scale): No
- What you personally verified and how: Protocol fixtures and mock BLE transport tests.
- Edge cases checked: split and concatenated frames, truncated weight recovery, real-record truncation at cuts 11-16 preserving the following frame, embedded headers, unknown records, settings length 9, button length 6, and combined flags.
- What you did not verify: physical LUNAR-E01ED7 or Pyxis hardware.

### Evidence

- [x] Test output (focused tests and analyzer)
- [x] Log snippets
- [ ] Screenshot / recording (UI changes)
- [ ] curl / websocat output (API changes)

## Compatibility & Migration

- Backward compatible? Yes
- Config / env changes needed? No
- Database migration needed? No
- If any No or Yes, explain exact steps: None.

## Risks & Mitigations

- Risk: Acaia firmware variants may still use unobserved record layouts.
  - Mitigation: frame-bounded parsing, strict non-weight record validation, and physical validation before merge.

### CODE REVIEW

File: lib/src/models/device/impl/acaia/acaia_scale.dart [hunk 1]
```diff
@@ -406,17 +432,114 @@ class AcaiaScale implements Scale {
...
+      5 => _metadataLength,
+      11
+          when frame.length > _metadataLength + 2 &&
+              frame[_metadataLength + 2] == 5 =>
+        _metadataLength + 3,
+      _ => -1,
     };
+    if (weightOffset < 0) return false;
+    final nextHeader = _findHeader(frame, start: 2);
+    return nextHeader == weightOffset + _weightBodyLength - 1;
```
Comment (tadelv) [inline lib/src/models/device/impl/acaia/acaia_scale.dart]: Non-blocking: this recovery check only recognizes the next header when it lands in the missing final weight-flag slot. If a declared multi-record frame is cut after all six weight bytes—so only its trailing record/checksum is missing—the following frame begins at `weightOffset + _weightBodyLength`. The parser can then fill the first frame with bytes from the next one, accept the complete weight, and lose that next frame. Could we add a regression test using a truncated `_realWeightFrame` followed by a frame with a different weight, and preserve the second frame during resync?

File: lib/src/models/device/impl/acaia/acaia_scale.dart [hunk 2]
```diff
@@ -62,7 +67,7 @@ class AcaiaScale implements Scale {
     0x01,
     0x02,
     0x02,
-    0x01,
+    0x05,
```
Comment (tadelv) [inline lib/src/models/device/impl/acaia/acaia_scale.dart:70]: Non-blocking scope note: changing this byte from `0x01` to `0x05` also changes the requested notification interval, independently of the declared-length parser fix. Since real hardware verification will need to come from beta users, could the PR summary explicitly call this out and cite the reference capture/implementation supporting `0x05`? Splitting it would also be reasonable, but I do not think this needs to block the parser fix.

**Review by tadelv — COMMENTED** [review]
Two non-blocking observations from a light review. Neither is intended as a request for changes.

**Review by tadelv — APPROVED** [review]
